apps/aechannel/src/aesc_client_reconnect_tx.erl

```diff
@@ -0,0 +1,185 @@
+-module(aesc_client_reconnect_tx).
+
+-behavior(aetx).
+-behavior(aesc_signable_transaction).
+
+%% Behavior API
+-export([
+          new/1
+        , type/0
+        , fee/1
+        , gas/1
+        , ttl/1
+        , nonce/1
+        , origin/1
+        , check/3
+        , process/3
+        , signers/2
+        , version/1
+        , serialization_template/1
+        , serialize/1
+        , deserialize/2
+        , for_client/1
+        , valid_at_protocol/2
+        ]).
+
+-export([
+          channel_id/1
+        , channel_pubkey/1
+        , role/1
+        , round/1
+        , state_hash/1
+        ]).
+
+-define(CHANNEL_CLIENT_RECONNECT_TX_TYPE, channel_client_reconnect_tx).
+-define(CHANNEL_CLIENT_RECONNECT_TX_FEE, 0).  % off-chain
+
+-define(INITIAL_VSN, 1).
+
+-type vsn() :: non_neg_integer().
+
+-record(channel_client_reconnect_tx,
+        {
+          channel_id   :: aeser_id:id()
+        , role         :: initiator | responder
+        , pub_key      :: aeser_id:id()
+        }).
+
+-opaque tx() :: #channel_client_reconnect_tx{}.
+
+-export_type([tx/0]).
+
+
+-spec new(map()) -> {ok, aetx:tx()}.
+new(#{ channel_id := ChannelId
+     , role       := Role
+     , pub_key    := Pubkey }) ->
+    channel = aeser_id:specialize_type(ChannelId),
+    account = aeser_id:specialize_type(Pubkey),
+    true = (Role == initiator) orelse (Role == responder),
+    Tx = #channel_client_reconnect_tx{
+            channel_id = ChannelId
+          , role       = Role
+          , pub_key    = Pubkey },
+    {ok, aetx:new(?MODULE, Tx)}.
+
+type() ->
+    ?CHANNEL_CLIENT_RECONNECT_TX_TYPE.
+
+-spec fee(tx()) -> non_neg_integer().
+fee(#channel_client_reconnect_tx{}) ->
+    %% This tx should never hit the mempool or any block
+    ?CHANNEL_CLIENT_RECONNECT_TX_FEE.
+
+-spec gas(tx()) -> non_neg_integer().
+gas(#channel_client_reconnect_tx{}) ->
+    0.
+
+-spec ttl(tx()) -> aetx:tx_ttl().
+ttl(#channel_client_reconnect_tx{}) ->
+    0.
+
+-spec nonce(tx()) -> non_neg_integer().
+nonce(#channel_client_reconnect_tx{}) ->
+    0.
+
+-spec origin(tx()) -> aec_keys:pubkey().
+origin(#channel_client_reconnect_tx{pub_key = Pubkey}) ->
+    aeser_id:specialize(Pubkey, account).
+
+-spec check(tx(), aec_trees:trees(), aetx_env:env()) -> {ok, aec_trees:trees()} | {error, term()}.
+check(#channel_client_reconnect_tx{
+        channel_id = _ChannelId
+      , role       = _Role
+      , pub_key    = _Pubkey }, Trees, _Env) ->
+    {ok, Trees}.
+
+-spec process(tx(), aec_trees:trees(), aetx_env:env()) -> {ok, aec_trees:trees()}.
+process(#channel_client_reconnect_tx{}, _Trees, #{}) ->
+    error(off_chain_tx).
```

Comment:
Does this mean that the TX can hit the mem pool but the miners won't include this TX inside a block?

Reply:
This is the same as for `aesc_offchain_tx`. Also, the fee and ttl are zero, so chances that miners will pick it up even if it would be pushed to the mempool should be 0.